src/steps/resource-manager/authorization/converters.ts

```diff
@@ -35,3 +48,33 @@ export function createRoleDefinitionEntity(
   };
   return entity;
 }
+
+export function createRoleAssignmentRelationship(
+  webLinker: AzureWebLinker,
+  roleAssignment: RoleAssignment,
+): Relationship {
+  const targetType = getJupiterTypeForPrincipalType(
+    roleAssignment.principalType as PrincipalType,
+  );
+  const roleDefinitionKey = (roleAssignment.roleDefinitionId as string).replace(
+    roleAssignment.scope as string,
+    '',
+  );
+  return createIntegrationRelationship({
+    _class: ROLE_ASSIGNMENT_RELATIONSHIP_CLASS,
+    _type: createRoleAssignmentRelationshipType(targetType),
+    properties: {
+      ...convertProperties(roleAssignment),
+      webLink: webLinker.portalResourceUrl(roleAssignment.id),
+    },
+    _mapping: {
+      relationshipDirection: RelationshipDirection.FORWARD,
+      sourceEntityKey: roleDefinitionKey,
+      targetFilterKeys: ['id'],
+      targetEntity: {
+        id: roleAssignment.principalId,
+      },
```

Comment:
```suggestion
      targetFilterKeys: [['_type', 'id']],
      targetEntity: {
        _type: targetEntityType,
        id: roleAssignment.principalId,
      },
```

I think this should include the `_type` of the entity, such as `azure_user` or `azure_group` or `azure_application` etc. This will ensure that when the mapper creates the target entity, it will have `_type` and `id` on it. In fact, in cases where we know how the `_key` is constructed, we should have all of `_type`, `_key`, and `id` on the target.

This will ensure the mapper relates to the existing `azure_user`, for example.

Reply:
Totally agree

Reply:
Adding `_key` and `_type` would allow us to use the `MappedRelationshipOptions` interface for `createRelationship()`, which is a safer option than the `MappedRelationshipLiteralOptions` interface.